File: java/src/main/java/org/knguyenc/sorting/BubbleSort.java
package org.knguyenc.sorting;

/**
 * Java implementation of bubble sort algorithm
 *
 * @author Khoa Nguyen
 */
public class BubbleSort extends AbstractSortingAlgorithm {

    @Override
    public <T extends Comparable<T>> void sort(T[] array) {
        for (int i = 0; i < array.length - 1; i++)
            for (int j = i + 1; j < array.length; j++)
                if (gt(array[i], array[j]))
                    swap(array, i, j);
    }
}

File: src/sorting/bubble_sort.rs
pub fn sort<T: PartialOrd>(arr: &mut [T]) {
    for i in 0..arr.len() - 1 {
        for j in i + 1..arr.len() {
            if arr[i] > arr[j] {
                arr.swap(i, j);
            }
        }
    }
}
